python/TestHarness/RunParallel.py

```diff
@@ -234,19 +252,37 @@ def spinwait(self, time_to_wait=0.05):
         for tuple in self.jobs:
             if tuple != None:
                 (p, command, tester, start_time, f, slots) = tuple
-                if p.poll() != None or now > (start_time + float(tester.specs['max_time'])):
-                    # finish up as many jobs as possible, don't sleep until
-                    # we've cleared all of the finished jobs
-                    self.returnToTestHarness(job_index)
+                # Look for completed tests
+                if p.poll() != None:
+                    if not self.checkOutputReady(tester):
```

Comment:
This is the key with this PR. We test to make sure that the output is available and if not we leave the job in the jobs queue for awhile longer (self.default_reporting_time or 10 seconds).